### PR TITLE
Improve logging in friendly client

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
In cases where we have a registration conflict, we should print out what it is that the user was trying to register so that the user can compare against what's stored in Admin.

In the future, we should have Admin return a proper diff.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Adding logging.
Auto-formatted

## Tracking Issue
https://github.com/lyft/flyte/issues/340

## Follow-up issue
NA

### CODE REVIEW

**Comment by kumare3** [comment]
👍 LGTM

File: flytekit/clients/friendly.py [hunk 1]
```diff
@@ -353,12 +370,19 @@ def create_launch_plan(
...
+        create_request = _launch_plan_pb2.LaunchPlanCreateRequest(
+            id=launch_plan_identifer.to_flyte_idl(),
+            spec=launch_plan_spec.to_flyte_idl()
         )
+        try:
+            super(SynchronousFlyteClient, self).create_launch_plan(create_request)
+        except _grpc.RpcError as e:
+            # When Admin detects a conflicting entry or otherwise invalid request during registration, it returns an
+            # invalid argument code. We should print out the thing we were trying to create in these cases.
+            if e.code() == _grpc.StatusCode.INVALID_ARGUMENT:
```
Comment (kumare3) [inline flytekit/clients/friendly.py:382]: why not dump all other error codes here? This way we could probably make the error message better?

**Comment by wild-endeavor** [comment]
Did a bit of a compromise.  @matthewphsmith ptal - I moved it to the raw client, and i'm using a decorator, but not the same decorator.  I think it's better to have a separate decorator than pick through *args/**kwargs.
